aws-rds-dbinstance/src/test/java/software/amazon/rds/dbinstance/CreateHandlerTest.java

```diff
@@ -1444,4 +1448,383 @@ public void handleRequest_RestoreDBInstanceFromSnapshot_HandleException(
+    }
+
+    @Test
+    public void handleRequest_RestoreDBInstanceToPointInTime_TriggeredBy_TargetDBInstanceIdentifier() {
+        final RestoreDbInstanceToPointInTimeResponse restoreResponse = RestoreDbInstanceToPointInTimeResponse.builder().build();
+        when(rdsProxy.client().restoreDBInstanceToPointInTime(any(RestoreDbInstanceToPointInTimeRequest.class)))
+                .thenReturn(restoreResponse);
+
+        final CallbackContext context = new CallbackContext();
+        context.setCreated(false);
+        context.setUpdated(true);
+        context.setRebooted(true);
+        context.setUpdatedRoles(true);
+
+
+        test_handleRequest_base(
+                context,
+                () -> DB_INSTANCE_ACTIVE,
+                () -> RESOURCE_MODEL_RESTORING_TO_POINT_IN_TIME.toBuilder()
+                        .targetDBInstanceIdentifier(TARGET_DB_INSTANCE_IDENTIFIER_NON_EMPTY)
+                        .build(),
+                expectSuccess()
+        );
+
+        verify(rdsProxy.client(), times(1)).restoreDBInstanceToPointInTime(any(RestoreDbInstanceToPointInTimeRequest.class));
+        verify(rdsProxy.client(), times(2)).describeDBInstances(any(DescribeDbInstancesRequest.class));
+    }
+
+    @Test
+    public void handleRequest_RestoreDBInstanceToPointInTime_NullIdentifiers() {
+        final RestoreDbInstanceToPointInTimeResponse restoreResponse = RestoreDbInstanceToPointInTimeResponse.builder().build();
+        when(rdsProxy.client().restoreDBInstanceToPointInTime(any(RestoreDbInstanceToPointInTimeRequest.class)))
+                .thenReturn(restoreResponse);
+
+        final CallbackContext context = new CallbackContext();
+        context.setCreated(false);
+        context.setUpdated(true);
+        context.setRebooted(true);
+        context.setUpdatedRoles(true);
+
+        test_handleRequest_base(
+                context,
+                () -> DB_INSTANCE_ACTIVE,
+                () -> RESOURCE_MODEL_RESTORING_TO_POINT_IN_TIME.toBuilder()
+                        .useLatestRestorableTime(USE_LATEST_RESTORABLE_TIME_YES)
+                        .build(),
+                expectSuccess()
+        );
+
+        ArgumentCaptor<RestoreDbInstanceToPointInTimeRequest> captor = ArgumentCaptor.forClass(RestoreDbInstanceToPointInTimeRequest.class);
+
+        verify(rdsProxy.client(), times(1)).restoreDBInstanceToPointInTime(captor.capture());
+        verify(rdsProxy.client(), times(2)).describeDBInstances(any(DescribeDbInstancesRequest.class));
+
+        // Both identifiers are passed as null. DBInstanceIdentifier will be random and the same value will be used for TargetDBInstanceIdentifier
+        // All of these InstanceIdentifier tests are non-perfect since we can't tell the value of DBInstanceIdentifier. So this is only half the picture
+        Assertions.assertThat(captor.getValue().targetDBInstanceIdentifier()).isNotNull();
```

Comment:
There is an even better method: `assertThat("foo").isNotBlanc()`, and I believe we are even more interested in this assertion here.